RevenueCatUI/Templates/V2/PaywallsV2View.swift

```diff
@@ -237,23 +237,47 @@ private struct LoadedPaywallsV2View: View {
     }
 
     var body: some View {
-        VStack(spacing: 0) {
-            ComponentsView(
-                componentViewModels: [paywallState.componentViewModel],
-                onDismiss: self.onDismiss
+        GeometryReader { proxy in
+            VStack(spacing: 0) {
+                ComponentsView(
+                    componentViewModels: [.root(paywallState.rootViewModel)],
+                    onDismiss: self.onDismiss
+                )
+            }
+            // If the first view in the first stack is an image,
+            // we will ignore safe area pass the safe area insets in to environment
+            // If the image is in a ZStack, the ZStack will push non-images
+            // down with the inset
```

Comment:
Talking about an edge case here but what happens if the parent of the image (being the first view within the first stack) is an HStack with an image + some text to the side? I imagine in that situation we might not want to ignore the safe area... So maybe this should only be done if the parent of the image is either a VStack or a ZStack?

Reply:
Good point! I just changed the logic so that the image needs to have a `width` of `fill` for this to work so... I _think_ that should cover the `HStack` scenario 🤔